File: plugin-build/src/test/fixtureSources/androidxSqliteDriver/InferredLocal.java
package com.example;

import androidx.room.RoomDatabase;
import androidx.sqlite.SQLiteDriver;
import androidx.sqlite.driver.bundled.BundledSQLiteDriver;

/**
 * A {@code checkcast} to {@code SQLiteDriver} sits immediately before the {@code setDriver} call
 * site (here forced via an {@code Object}-typed local). This mirrors the bytecode kotlinc emits for
 * an inferred-type local and verifies the visitor still inserts the wrap when a {@code checkcast}
 * precedes the invocation.
 */
public class InferredLocal {

  public void build() {
    Object driver = new BundledSQLiteDriver();
    new RoomDatabase.Builder().setDriver((SQLiteDriver) driver);
  }
}

File: plugin-build/src/test/fixtureSources/androidxSqliteDriver/InferredLocal.kt
import androidx.room.RoomDatabase
import androidx.sqlite.driver.bundled.BundledSQLiteDriver

/**
 * Kotlin reference showing the idiomatic equivalent of [InferredLocal.java]. The compiled `.class`
 * used by tests comes from the Java source; this file is kept for documentation only (it shows the
 * `val driver = BundledSQLiteDriver()` pattern that makes kotlinc emit a `checkcast` to
 * `SQLiteDriver` at the `setDriver` call site).
 */
class InferredLocal {
  fun configure(builder: RoomDatabase.Builder<RoomDatabase>) {
    val driver = BundledSQLiteDriver()
    builder.setDriver(driver)
  }
}
